### PR TITLE
Add diagrams for each use case

### PR DESCRIPTION
**Pull request by karansohi**
- Add mermaid for each use case with explanation below
- Add overview sessions to give high level explanation of use-case
- Add high-level real-world use-case scenario for each use-case
- rename basic service protection to protection
- rename workload-prioritization to prioritization
- fix broken links

##### Checklist

- [ ] Tested in playground or other setup
- [ ] Screenshot (Grafana) from playground added to PR for 15+ minute run
- [x] Documentation is changed or added
- [ ] Tests and/or benchmarks are included
- [ ] Breaking changes

### Summary by OpenAI

**Documentation:**
- Updated and improved use case documentation
- Added diagrams, explanations, and real-world scenarios
- Fixed broken links and renamed files

> 🎉 A new dawn for docs, we've made them shine,
> With diagrams and use cases, they're now so fine.
> Broken links are fixed, and files renamed,
> Our users' understanding will never be the same! 🚀

### CODE REVIEW

File: docs/content/use-cases/alerting/alerting.md [hunk 1]
```diff
@@ -7,7 +7,44 @@ sidebar_position: 5
...
+The graph depicts the process of the Agent writing any overload detection to
+Prometheus. These metrics are queried by the Controller and relayed to the Alert
+Manager. The Controller includes a signal processing mechanism that can
+distinguish false positives by comparing current latency with the setpoint.
+
+:::note
+
+Aperture facilitates the observation of health signals from various services.
+For instance, service protection can also be implemented based on the health
+observation of an upstream service in relation to a downstream service.
```
Comment (tanveergill) [inline docs/content/use-cases/alerting/alerting.md:35]: This is supposed to be an overview of Alerting in Aperture. Not sure why it's focusing on a particular use-case (detecting overload)

File: docs/content/introduction.md [hunk 2]
```diff
@@ -119,12 +119,14 @@ in defining responses to different situations, such as deviations from
...
-additional Flow Labels based on request metadata, collectively enriching the
-policy's operational capability.
+
+- [**Control Point**](./concepts/flow-control/selector.md): Needed to guide flow
+  control based decisions.
+- [**Selector**](./concepts/flow-control/selector.md): Needed to select the flow
+  that the policy should consider.
+- [**Classifier**](./concepts/flow-control/resources/classifier.md): Create
+  additional Flow Labels based on request metadata, collectively enriching the
+  policy's operational capability.
```
Comment (sudhanshu456) [inline docs/content/introduction.md:129]: @karansohi  Can you help me understand the reasoning behind putting this here? Also, we could be talking about it in your first policy doc.

**Review by sudhanshu456 — COMMENTED** [review]
The rest of the changes look good to me!

File: docs/content/introduction.md (hunk 2 above)
Comment (karansohi) [reply docs/content/introduction.md:129]: I was just fixing an indentation but yeah I agree it can be removed. Right? @tanveergill

**Review by tanveergill — APPROVED** [review]

**Pull request merged**